api/handlers.go

```diff
@@ -747,6 +754,8 @@ func postContainersStart(c *context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	container.Config.HostConfig = hostConfig
```

Comment:
Or, as we did before, don't even save the config on start, just pass it via are to dockerclient, I think it makes sense